contracts/SnappAuction.sol

```diff
@@ -187,6 +195,51 @@ contract SnappAuction is SnappBase {
         }
     }
 
+    function auctionSolutionBid(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 proposedStateRoot,
+        uint proposedObjectiveValue
+        // bytes32 _orderHash,
+        // bytes32 _standingOrderIndex          // should ensure acting on correct order set
+    ) public {
+        require(
+            slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0,
+            "Previous auction not yet resolved!"
+        );
+        // Verify state root and order hash before continuing.
+        require(coreData.stateRoots[stateIndex()] == _currStateRoot, "Incorrect state root");
+        // TODO - verify acting on correct order hash.
+        // require(
+        //     calculateOrderHash(slot, _standingOrderIndex) == _orderHash,
+        //     "Order hash doesn't agree"
+        // );
+
+        // Ensure that auction batch is inactive, unprocessed and in correct phase for bidding
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");  // This is a bit redundant.
+        require(slot != MAX_UINT && slot <= auctionIndex, "Requested order slot does not exist");
```

Comment:
nit: auction slot instead of order slot.

Reply:
```suggestion
        require(slot != MAX_UINT && slot <= auctionIndex, "Requested auction slot does not exist");
```